ClassTranscribeServer/Controllers/StaticFileController.cs

```diff
@@ -52,33 +55,41 @@ public async Task<ActionResult> GetFile(string? relpath)
             {
                 return NotFound();
             }
-            // See Authorization.cs
-            string? referer = HttpContext.Request.Headers["Referer"];
-           
-            if(string.IsNullOrEmpty(referer) || !  referer.Contains( HttpContext.Request.Host.Host,System.StringComparison.InvariantCultureIgnoreCase)) {
-                return NotFound();
+            // Require Authenticated user
+            if( ! checkFileAccess(urlsubpath, User ) ) {
+                return Unauthorized();
```

Comment:
Is that gonna return 403?

Reply:
That's 401. but I'll update it in a moment to be Forbid() ie. 403. Hold on...